src/riverroute/RtmMod.F90

```diff
@@ -1,2844 +1,2407 @@
+         endif
+
+         call t_stopf('mosartr_budget')
+      endif  ! budget_check
+
+      !-----------------------------------
+      ! Write out MOSART history file
+      !-----------------------------------
+
+      call t_startf('mosartr_hbuf')
+      call RtmHistFldsSet()
+      call RtmHistUpdateHbuf()
+      call t_stopf('mosartr_hbuf')
+
+      call t_startf('mosartr_htapes')
+      call RtmHistHtapesWrapup( rstwr, nlend )
+      call t_stopf('mosartr_htapes')
+
+      !-----------------------------------
+      ! Write out MOSART restart file
+      !-----------------------------------
+
+      if (rstwr) then
+         call t_startf('mosartr_rest')
+         filer = RtmRestFileName(rdate=rdate)
+         call RtmRestFileWrite( filer, rdate=rdate )
+         call t_stopf('mosartr_rest')
+      end if
+
+      !-----------------------------------
+      ! Done
+      !-----------------------------------
+
+      first_call = .false.
+
+      call t_stopf('mosartr_tot')
+
+   end subroutine MOSART_run
+
+   !-----------------------------------------------------------------------
+
+   subroutine MOSART_FloodInit(frivinp, begr, endr, fthresh, evel )
+
+      ! Arguments
+      character(len=*) , intent(in)  :: frivinp
+      integer          , intent(in)  :: begr, endr
+      real(r8)         , intent(out) :: fthresh(begr:endr)
+      real(r8)         , intent(out) :: evel(begr:endr,nt_rtm)
+
+      ! Local variables
+      real(r8), pointer  :: rslope(:)
+      real(r8), pointer  :: max_volr(:)
+      integer , pointer  :: compdof(:) ! computational degrees of freedom for pio
+      integer            :: nt,n,cnt   ! indices
+      logical            :: readvar    ! read variable in or not
+      integer            :: ier        ! status variable
+      integer            :: dids(2)    ! variable dimension ids
+      type(file_desc_t)  :: ncid       ! pio file desc
+      type(var_desc_t)   :: vardesc    ! pio variable desc
+      type(io_desc_t)    :: iodesc     ! pio io desc
+      character(len=256) :: locfn      ! local file name
+
+      ! MOSART Flood variables for spatially varying celerity
+      real(r8) :: effvel(nt_rtm) = 0.7_r8   ! downstream velocity (m/s)
+      real(r8) :: min_ev(nt_rtm) = 0.35_r8  ! minimum downstream velocity (m/s)
+      real(r8) :: fslope = 1.0_r8           ! maximum slope for which flooding can occur
+      character(len=*),parameter :: subname = '(MOSART_FloodInit) '
+      !-----------------------------------------------------------------------
+
+      allocate(rslope(begr:endr), max_volr(begr:endr), stat=ier)
+      if (ier /= 0) call shr_sys_abort(subname // ' allocation ERROR')
+
+      ! Assume that if SLOPE is on river input dataset so is MAX_VOLR and that
+      ! both have the same io descriptor
+
+      call getfil(frivinp, locfn, 0 )
+      call ncd_pio_openfile (ncid, trim(locfn), 0)
+      call pio_seterrorhandling(ncid, PIO_BCAST_ERROR)
```

Comment:
This demonstrates correct use of PIO_BCAST_ERROR - the method is set in line 2324, the error is checked in 2326 and the method is reset in 2332.

Reply:
Thanks for pointing this out.